pkg/apis/healthchecking/v1alpha1/machinehealthcheck_types.go

```diff
@@ -36,6 +38,9 @@ type MachineHealthCheckList struct {
 
 // MachineHealthCheckSpec defines the desired state of MachineHealthCheck
 type MachineHealthCheckSpec struct {
+	// RemediationStrategy to use in case of problem detection
+	// default is machine deletion
+	RemediationStrategy RemediationStrategyType `json:"remediationStrategy"`
```

Comment:
If this is optional and no value means use the default behavior, should this be a pointer with `omitempty`? Maybe it doesn't matter.

Also, if we have the ability, it might be nice to annotate this for validation. Something like:
```
// +kubebuilder:validation:Enum=reboot
```

Reply:
I'll explore the validation, no need to be a pointer since no need to difference between nil, "" and "value". We just have 2 cases ""/"value"